### PR TITLE
Disable POMELO on big-endian machines

### PR DESCRIPTION
**Pull request by kholia**
This is related to https://github.com/magnumripper/JohnTheRipper/issues/2868.

### CODE REVIEW

**Comment by magnumripper** [comment]
It's definitely not the end of the world but I kind of dislike this. Is the problems caused by excessive optimizations? Keeping some basic reference implentation (no SIMD, whatever) would be great.

**Comment by kholia** [comment]
AFAIK POMELO was never deployed / used widely (anywhere?). It should be OK to drop it.

**Comment by magnumripper** [comment]
I know that, I still want it (just like I want "SHA-0").

But obviously we can't put too much effort in it now. We could move it to `unused` instead of dropping it completely!

**Comment by solardiz** [comment]
IIRC, endianness issues with POMELO were also reported in PHC. I don't recall if there ever was a fixed version. If we want to keep this format, the first step would be to see if we're at the latest revision submitted to PHC or not. The latest is v3 listed here: https://password-hashing.net/submissions.html

Maybe v3 just works on big-endian. If so, great. If not, we can `#if ARCH_LITTLE_ENDIAN` this entire format (does the pluggable formats infrastructure support that?), not to waste our own effort on fixing its code.

**Comment by magnumripper** [comment]
> we can #if ARCH_LITTLE_ENDIAN this entire format (does the pluggable formats infrastructure support that?)

Yes, that's trivial! @kholia please do so instead.

**Comment by solardiz** [comment]
Oh, we have a comment saying we're at v2. So try v3?

And I forgot those hashes were in Hash Runner. That may be a reason to keep the format at least in `unused`, in case someone wants to relive that contest. ;-) OTOH, that would require v3 to be compatible with v2 - this is probably the case, but I don't recall it for sure. So perhaps swap v3 in place of our v2 and see if the test still passes (with our existing test vectors) on LE and if things start working on BE?

**Comment by kholia** [comment]
I checked and our code is same as the one in `POMELO-v3.tar.gz`.

The POMELO code has a comment saying `This codes gives the C implementation of POMELO on 64-bit platform (little-endian)`. So I guess I can't complain too much when the code fails to run on big-endian systems.

File: src/pomelo_fmt_plug.c [hunk 1]
```diff
@@ -3,6 +3,9 @@
  * contest by Dhiru Kholia.
  */
```
Comment (solardiz) [inline src/pomelo_fmt_plug.c]: `#include "arch.h"`

File: src/pomelo_fmt_plug.c [hunk 2]
```diff
@@ -3,6 +3,9 @@
...
 
+// Enable this format only on little-endian systems
+#ifdef ARCH_LITTLE_ENDIAN
```
Comment (solardiz) [inline src/pomelo_fmt_plug.c]: `#if ARCH_LITTLE_ENDIAN`

We're using these `arch.h` macros as 0 vs. 1, not as defined vs. undefined.

File: src/pomelo_plug.c [hunk 3]
```diff
@@ -1,3 +1,5 @@
+// NOTE: This code is based on https://password-hashing.net/submissions/POMELO-v3.tar.gz archive.
+
 // PHC submission:  POMELO v2
```
Comment (solardiz) [inline src/pomelo_plug.c]: Does the code from the v3 tarball also still have "v2" in this comment, or do we need to update the comment?

Comment (kholia) [reply src/pomelo_plug.c]: It still has the "v2" in the comment.

File: src/pomelo_fmt_plug.c (hunk 2 above)
Comment (kholia) [reply src/pomelo_fmt_plug.c]: It seems that I did a sloppy job with this PR ;(

File: src/pomelo_fmt_plug.c (hunk 1 above)
Comment (kholia) [reply src/pomelo_fmt_plug.c]: Maybe this is not needed. 

In the output of `make` I see `gcc -DAC_BUILT ... -DARCH_LITTLE_ENDIAN=1 ...`. It seems that the code will "know" `ARCH_LITTLE_ENDIAN` value even without including `arch.h`.

Comment (solardiz) [reply src/pomelo_fmt_plug.c]: Oh, that's jumbo-specific. I didn't recall. Anyway, I suggest that you do include `arch.h`, such as for `Makefile.legacy` to work.

**Comment by solardiz** [comment]
I vaguely recall there was also an AVX2 version of the POMELO code, included in one of those upstream updates. Should we add that? %-)

**Comment by kholia** [comment]
Hmm. I guess the POMELO format got lucky recently ;)

I will add the AVX2 version soon.

**Pull request merged**